lgc/builder/YCbCrConverter.cpp

```diff
@@ -0,0 +1,917 @@
+
+        imageOpChroma = wrappedSample(wrappedSampleInfo);
+      } else // SamplerFilter::Linear
+      {
+        if (m_metaData.word1.ySubSampled) {
+          imageOpChroma = reconstructLinearXYChromaSample(xyChromaInfo);
+        } else {
+          imageOpChroma = reconstructLinearXChromaSample(xChromaInfo);
+        }
+      }
+    } else {
+      if (m_metaData.word1.xSubSampled) {
+        wrappedSampleInfo.coordI =
+            calculateImplicitChromaUV(static_cast<ChromaLocation>(m_metaData.word1.xChromaOffset), m_coordU);
+      }
+
+      if (m_metaData.word1.ySubSampled) {
+        wrappedSampleInfo.coordJ =
+            calculateImplicitChromaUV(static_cast<ChromaLocation>(m_metaData.word1.yChromaOffset), m_coordV);
+      }
+
+      imageOpChroma = wrappedSample(wrappedSampleInfo);
+    }
+  } else // lumaFilter == SamplerFilter::Linear
+  {
+    if (m_metaData.word0.forceExplicitReconstruct || !(m_metaData.word1.xSubSampled || m_metaData.word1.ySubSampled)) {
+      Value *lumaA = calculateUVoffset(m_coordU);
+      Value *lumaB = calculateUVoffset(m_coordV);
+      Value *subIPlusOne = m_builder->CreateFAdd(m_coordI, ConstantFP::get(m_builder->getFloatTy(), 1.0f));
+      Value *subJPlusOne = m_builder->CreateFAdd(m_coordJ, ConstantFP::get(m_builder->getFloatTy(), 1.0f));
+
+      if ((static_cast<SamplerFilter>(m_metaData.word1.chromaFilter) == SamplerFilter::Nearest) ||
+          !m_metaData.word1.xSubSampled) {
+        if (!m_metaData.word1.xSubSampled) {
+          wrappedSampleInfo.subsampledX = false;
+          wrappedSampleInfo.subsampledY = false;
+          imageOpChroma = wrappedSample(wrappedSampleInfo);
+        } else {
+          Value *subCoordI = m_coordI;
+          Value *subCoordJ = m_coordJ;
+          if (m_metaData.word1.xSubSampled) {
+            subCoordI = m_builder->CreateFDiv(m_coordI, ConstantFP::get(m_builder->getFloatTy(), 2.0));
+            subIPlusOne = m_builder->CreateFDiv(subIPlusOne, ConstantFP::get(m_builder->getFloatTy(), 2.0));
+          }
+
+          if (m_metaData.word1.ySubSampled) {
+            subCoordJ = m_builder->CreateFDiv(m_coordJ, ConstantFP::get(m_builder->getFloatTy(), 2.0));
+            subJPlusOne = m_builder->CreateFDiv(subJPlusOne, ConstantFP::get(m_builder->getFloatTy(), 2.0));
+          }
+
+          wrappedSampleInfo.coordI = subCoordI;
+          wrappedSampleInfo.coordJ = subCoordJ;
+          Value *coordTL = wrappedSample(wrappedSampleInfo);
+
+          wrappedSampleInfo.coordI = subIPlusOne;
+          Value *coordTR = wrappedSample(wrappedSampleInfo);
+
+          wrappedSampleInfo.coordJ = subJPlusOne;
+          Value *coordBR = wrappedSample(wrappedSampleInfo);
+
+          wrappedSampleInfo.coordI = subCoordI;
+          Value *coordBL = wrappedSample(wrappedSampleInfo);
+
+          imageOpChroma = bilinearBlend(lumaA, lumaB, coordTL, coordTR, coordBL, coordBR);
+        }
+      } else // vk::VK_FILTER_LINEAR
```

Comment:
Maybe replace "vk::XXX" with others. LGC comments ought to not have VK enum descriptions